distributed/tests/test_worker.py

```diff
@@ -1180,3 +1180,19 @@ def test_wait_for_outgoing(c, s, a, b):
     ratio = aa / bb
 
     assert 1 / 3 < ratio < 3
+
+
+@gen_cluster(client=True, ncores=[('127.0.0.1', 1)] * 20, timeout=30)
+def test_avoid_oversubscription(c, s, *workers):
+    np = pytest.importorskip('numpy')
+    x = c.submit(np.random.random, 1000000, workers=[workers[0].address])
+    yield wait(x)
+
+    with dask.config.set({'distributed.worker.max-connections': 1}):
+        futures = [c.submit(len, x, pure=False, workers=[w.address])
+                   for w in workers[1:]]
+
+        yield wait(futures)
+
+    assert len(workers[0].outgoing_transfer_log) < 18
```

Comment:
Where does the 18 and the 3 come from? Where they measured from empirical runs?

Reply:
Two issues:

1.  We may start to run out of RAM on travis with 1e8 * 20 bytes
2.  Compression will make the transfers too fast

I'm not very concerned about the cost of creating the random array the first time.  I don't think that this will affect the number of secondary worker-to-worker transfers.  However I may not fully understand your meaning.